### DISCUSSION TITLE
Data Protection (Germany)

### DISCUSSION THREAD

**Discussion by jlorezz** [vercel/community#58] [Help] [answered]
How is Vercel handling the data protection (we are currently based around Germany).

**Comment by leerob** [comment, marked as answer]
Hey! Could you be a bit more specific?

More info here: https://vercel.com/security

**Reply by jlorezz** [reply]
Data processing, how long you store the data, how the data is used, etc. Germany might change the law to only allow storing data for 30d (This hasn't entirely gone through yet but is still under review)

**Reply by jlorezz** [reply]
On which servers is it hosted?  (Europe etc.)

**Reply by leerob** [reply]
You can see all of the regions here: https://vercel.com/docs/concepts/edge-network/regions

And you can block countries or IPs using Edge Functions if necessary: https://github.com/vercel/examples/tree/main/edge-functions/geolocation-country-block
